lib/galaxy/tool_util/linters/datatypes.py

```diff
@@ -78,6 +78,14 @@ def lint(cls, tool_source: "ToolSource", lint_ctx: "LintContext"):
         for attrib in ["format", "ftype", "ext"]:
             for elem in tool_xml.findall(f".//*[@{attrib}]"):
                 formats = elem.get(attrib, "").split(",")
+                if "auto" in formats:
```

Comment:
I think we should also check that for `collection`, `data`, `discover_datasets` and `when` elements, only one format is specified.
I've checked on tools-iuc, and there is only one counter case, which surely seems like a bug: https://github.com/galaxyproject/tools-iuc/blob/main/tools/pairtools/parse.xml#L46

```suggestion
                if elem.tag in ("collection", "data", "discover_datasets", "when") and len(formats) > 1:
                    lint_ctx.error(
                        f"Only one format can be specified for a {elem.tag} element",
                        linter=cls.name(),
                        node=elem,
                    )
                if "auto" in formats:
```

Reply:
I also thought this in the first place, but this is covered in the xsd: https://github.com/galaxyproject/tools-iuc/blob/4aa44523d9df9bf1773ed724735120476d773a19/tools/pairtools/.lint_skip#L1 All `format`/`ftype`/`ext` attributes should be annotated as `Format` in the XSD if only a single format is allowed and `FormatList` otherwise.

Reply:
Thanks, that's great! I've added a suggestion to mention that in a comment.